datafusion/sqllogictest/test_files/udf_preimage.slt

```diff
@@ -0,0 +1,170 @@
+# Licensed to the Apache Software Foundation (asF) under one
```

Comment:
I haven't reviewed the test coverage yet, but can we please ensure we have added all the corresponding tests from duckdb (both initial PR and the fix):
- https://github.com/duckdb/duckdb/pull/18457
- https://github.com/duckdb/duckdb/pull/19628


We can probably copy most of the tests from the following places (they use sqllogictests as well)
https://github.com/duckdb/duckdb/blob/main/test/optimizer/date_trunc_simplification_icu.test
https://github.com/duckdb/duckdb/blob/main/test/optimizer/date_trunc_simplification.test